client/dashboard/domains/domain-contact-details/contact-form.tsx

```diff
@@ -0,0 +1,207 @@
+import { useMutation, useQuery } from '@tanstack/react-query';
+import {
+	ExternalLink,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+	__experimentalText as Text,
+	Button,
+	Card,
+	CardBody,
+} from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { DataForm, Field, isItemValid } from '@wordpress/dataviews';
+import { createInterpolateElement } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { useMemo, useState } from 'react';
+import { countryListQuery, statesListQuery } from '../../app/queries/domain-supported-contries';
+import { domainWhoisMutation } from '../../app/queries/domain-whois';
+import InlineSupportLink from '../../components/inline-support-link';
+import Notice from '../../components/notice';
+import {
+	validateDomainWhois,
+	type ContactValidationRequestContactInformation,
+	type DomainContactDetails,
+} from '../../data/domain-whois';
+import { getContactFormFields } from './contact-form-fields';
+
+interface ContactFormProps {
+	domainName: string;
+	initialData?: DomainContactDetails;
+	onSubmit?: ( data: DomainContactDetails ) => void;
+	onCancel?: () => void;
+	errors?: Partial< Record< keyof DomainContactDetails, string > >;
+}
+
+export default function ContactForm( {
+	domainName,
+	initialData,
+	onSubmit,
+	onCancel,
+}: ContactFormProps ) {
+	const { createSuccessNotice, createErrorNotice } = useDispatch( noticesStore );
+	const { data: countryList } = useQuery( countryListQuery() );
+	const [ formData, setFormData ] = useState< DomainContactDetails >(
+		initialData ?? { optOutTransferLock: false }
+	);
+	const selectedCountryCode = formData.countryCode ?? initialData?.countryCode ?? '';
+	const { data: statesList } = useQuery( statesListQuery( selectedCountryCode ) );
+	const updateMutation = useMutation( domainWhoisMutation( domainName ) );
+
+	const validateMutation = useMutation( {
+		mutationFn: ( formData: ContactValidationRequestContactInformation ) => {
```

Comment:
We should put this in the query layer as `domainWhoisValidateMutation()`, replacing the existing now-unused `domainWhoisValidateQuery()`.

Reply:
Fixed it in the past two commits